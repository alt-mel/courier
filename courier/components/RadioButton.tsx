import React from 'react';
import {
    StyleSheet, Pressable
  } from 'react-native';

import { Text, View } from '../components/Themed';

export default function RadioButton(props) {
    return (
        <Pressable style={styles.radioButton}>
        <View style={[{
          height: 24,
          width: 24,
          borderRadius: 12,
          borderWidth: 2,
          borderColor: '#ed706e',
          alignItems: 'center',
          justifyContent: 'center',

        }, props.style]}>{props.selected ?
          <View style={{
            height: 12,
            width: 12,
            borderRadius: 6,
            backgroundColor: '#ed706e'
          }} /> : null} </View><Text style={styles.selection}> {props.value}</Text></Pressable>
    );
}

const styles = StyleSheet.create({

    radioButton: {
      display: 'flex',
      flexDirection: 'row',
      marginBottom: 22
    },
    selection: {
      fontWeight: '600',
      lineHeight: 19,
      margin: 3,
      color: '#8C8C8C'
    }
  });
  