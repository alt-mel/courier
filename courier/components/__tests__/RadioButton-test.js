import * as React from 'react';
import renderer from 'react-test-renderer';

import RadioButton from '../RadioButton';

it('renders correctly', () => {
  const tree = renderer.create(<RadioButton selected="true" value="0 - 5 kg"/>).toJSON();

  expect(tree).toMatchSnapshot();
});

