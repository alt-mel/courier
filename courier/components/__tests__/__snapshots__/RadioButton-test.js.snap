// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 22,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#ed706e",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "backgroundColor": "#ed706e",
            "borderRadius": 6,
            "height": 12,
            "width": 12,
          },
        ]
      }
    />
     
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "#8C8C8C",
          "fontWeight": "600",
          "lineHeight": 19,
          "margin": 3,
        },
      ]
    }
  >
     
    0 - 5 kg
  </Text>
</View>
`;
